src/providers/ldap/sdap.c

```diff
@@ -148,6 +148,27 @@ static enum duplicate_t check_duplicate(struct sdap_attr_map *map,
     return NOT_FOUND;
 }
 
+/**
+ * @brief Add attributes to a map
+ *
+ * sdap_extend_map() will call talloc_realloc() on the second argument so the
+ * original storage location might change. The return value _map will always
+ * contain the current memory location which can be used with talloc_free()
+ * even if there is an error.
+ *
+ * @param[in] memctx      Talloc memory context
+ * @param[in] src_map     Original map, should not be accessed anymore
+ * @param[in] num_entries Number of entries in the original map
+ * @param[in] extra_attrs NULL-terminated array of extra attribute pairs
+ *                        sysdb_attr:ldap_attr
+ * @param[out] _map       New map
+ * @param[out] _new_size  Number of entries in the new map
+ *
+ * @return
+ *  - EOK                 success
+ *  - ENOMEM              memory allocation failed
+ *  - ERR_DUP_EXTRA_ATTR  sysdb attribute is already used
+ */
```

Comment:
I agree that the function API comment is useful. In my opinion -- since this function isn't static -- isn't better place for it in ```sdap.h``` file?